frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelineRuns.cy.ts

```diff
@@ -0,0 +1,313 @@
+    },
+    { runs: [mockRunKF({}), mockRunKF({ name: 'test-pipeline-2', id: 'test-pipeline-2' })] },
+  );
+  cy.intercept(
+    {
+      method: 'POST',
+      pathname: '/api/proxy/apis/v1beta1/pipelines',
+    },
+    mockPipelineKF({}),
+  );
+  cy.intercept(
+    {
+      method: 'POST',
+      pathname: '/api/proxy/apis/v1beta1/pipelines/test-pipeline/templates',
+    },
+    mockPipelinesVersionTemplateResourceKF(),
+  );
+  cy.intercept(
+    {
+      pathname:
+        '/api/k8s/apis/route.openshift.io/v1/namespaces/test-project/routes/ds-pipeline-pipelines-definition',
+    },
+    mockRouteK8sResource({ notebookName: 'ds-pipeline-pipelines-definition' }),
+  );
+  cy.intercept(
+    {
+      pathname: '/api/k8s/api/v1/namespaces/test-project/secrets/ds-pipeline-config',
+    },
+    mockSecretK8sResource({ name: 'ds-pipeline-config' }),
+  );
+  cy.intercept(
+    {
+      pathname: '/api/k8s/api/v1/namespaces/test-project/secrets/pipelines-db-password',
+    },
+    mockSecretK8sResource({ name: 'pipelines-db-password' }),
+  );
+  cy.intercept(
+    {
+      pathname: '/api/k8s/api/v1/namespaces/test-project/secrets/aws-connection-testdb',
+    },
+    mockSecretK8sResource({ name: 'aws-connection-testdb' }),
+  ),
+    cy.intercept(
+      {
+        pathname:
+          '/api/k8s/apis/datasciencepipelinesapplications.opendatahub.io/v1alpha1/namespaces/test-project/datasciencepipelinesapplications/pipelines-definition',
+      },
+      mockDataSciencePipelineApplicationK8sResource({}),
+    );
+  cy.intercept(
+    {
+      pathname: '/api/k8s/apis/kubeflow.org/v1/namespaces/test-project/notebooks',
+    },
+    mockK8sResourceList([mockNotebookK8sResource({})]),
+  );
+  cy.intercept(
+    {
+      pathname: '/api/k8s/apis/project.openshift.io/v1/projects',
+    },
+    mockK8sResourceList([mockProjectK8sResource({})]),
+  );
+};
+
+describe('Pipeline runs', () => {
+  describe('Test deleting runs', () => {
+    it('Test delete a single run from scheduled', () => {
+      initIntercepts();
+
+      globalPipelineRuns.visit('test-project');
+      globalPipelineRuns.findIsApiAvailable().should('exist');
+
+      globalPipelineRuns.findScheduledRunTab().click();
+      globalPipelineRuns
+        .getPipelineRunJobsTableRow('test-pipeline')
+        .find()
+        .findKebabAction('Delete')
+        .click();
+
+      scheduledRunDeleteModal.shouldBeOpen();
+      scheduledRunDeleteModal.findSubmitButton().should('be.disabled');
+      scheduledRunDeleteModal.findInput().type('test-pipeline');
+      scheduledRunDeleteModal.findSubmitButton().should('be.enabled');
+
+      cy.intercept(
+        {
+          method: 'POST',
+          pathname: '/api/proxy/apis/v1beta1/jobs/test-pipeline',
+        },
+        mockStatus(),
+      ).as('postJobPipeline');
+
+      scheduledRunDeleteModal.findSubmitButton().click();
+
+      cy.wait('@postJobPipeline').then((intercept) => {
+        expect(intercept.request.body).to.eql({
```

Comment:
This expectation is probably sufficient, but I was thinking it might be useful to also expect that the table no longer has visual of the deleted records after the request finishes. What are your thoughts on this?

Reply:
Unless this is an e2e test, then that won't be possible. We can only really tests against the request going out

Reply:
After the deletion and the modal closes, I'm referring to the outgoing fetch of runs/jobs that are made, which at that point would exclude the newly deleted run/job. So if that is mocked and we return a list of runs/jobs excluding that deleted one, we can verify that deleted run/job is no longer in the table (I think).

Reply:
added tests for this

Reply:
seems like the tests wont work when not run from the UI.

Reply:
this specific call

Reply:
wait, nvm i found a workaround